Gems/Atom/RPI/Code/Source/RPI.Public/Pass/AttachmentReadback.cpp

```diff
@@ -226,6 +225,12 @@ namespace AZ
                 {
                     m_bufferAttachmentByteSize = attachment->m_descriptor.m_buffer.m_byteCount;
                 }
+                m_readbackItems.push_back({});
+                auto& item = m_readbackItems.back();
+                for(uint32_t i = 0; i < RHI::Limits::Device::FrameCountMax; i++)
+                {
+                    item.m_readbackBufferArray.push_back(nullptr);
+                }
```

Comment:
I haven't used `Data::instance<>`, can you use the `resize` method with a nullptr? or would `resize()` init with nullptr types?
```suggestion
                item.m_readbackBufferArray.resize(RHI::Limits::Device::FrameCountMax, nullptr);
```

Reply:
I like your idea. Will give it a try.

Reply:
Done. Looks better.